### PR TITLE
chore(cypress e2e): Update Link checker for docs site

### PR DESCRIPTION
**Pull request by ErikCH**
#### Description of changes
Docs builds are failing because the number of links has grown. This will add the correct number of links

#### Issue #, if available
This will fix the issue with dev docs not building.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by 0618 — APPROVED** [review]

File: docs/cypress/e2e/links.cy.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ type EvtName =
   | 'RETURNING'
   | 'NO_HREF';
 let allLinks: string[] = [];
-const numberOfLinks = 119;
+const numberOfLinks = 121;
```
Comment (calebpollman) [inline docs/cypress/e2e/links.cy.ts:14]: @zhamujun Do we need to now update this manually every time a link is added? From [your PR](https://github.com/aws-amplify/amplify-ui/pull/2844): 
> Dynamic. When new links added or exist link changed, we don't have to update the tests.

**Review by calebpollman — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]

File: docs/cypress/e2e/links.cy.ts (hunk 1 above)
Comment (0618) [reply docs/cypress/e2e/links.cy.ts:14]: Yes, the links are dynamically tested, but the number of links has to be updated manually. There's no way for the test to know how many links there should be.

Comment (0618) [reply docs/cypress/e2e/links.cy.ts:14]: The description in my PR was not accurate. Let me change that...

Comment (0618) [reply docs/cypress/e2e/links.cy.ts:14]: Updated:
> Dynamic. When new links added or exist link changed, we don't have to update the tests. However, if a new page is added, we do need to update the number of pages being tested.

**Pull request merged**